stdlib/public/core/ArrayBuffer.swift

```diff
@@ -247,29 +247,27 @@ extension _ArrayBuffer {
       // Look for contiguous storage in the NSArray
       let nonNative = self._nonNative
       let cocoa = _CocoaArrayWrapper(nonNative)
-      let cocoaStorageBaseAddress = cocoa.contiguousStorage(self.indices)
-
-      if cocoaStorageBaseAddress != nil {
-        return _SliceBuffer(
-          owner: nonNative,
-          subscriptBaseAddress: UnsafeMutablePointer(cocoaStorageBaseAddress),
-          indices: subRange,
-          hasNativeBuffer: false)
-      }
-
-      // No contiguous storage found; we must allocate
-      let subRangeCount = subRange.count
-      let result = _ContiguousArrayBuffer<Element>(
+      guard let cocoaStorageBaseAddress = cocoa.contiguousStorage(self.indices) else {
```

Comment:
Same here, I don't think this is a win, since `guard` reads well for quick early exits.  In this case the quick exit path is returning a `_SliceBuffer`.